task_manager/task_manager.hh

```
@@ -0,0 +1,229 @@
+            if (_children.empty()) {
+                return make_ready_future<progress>(_status.progress);
+            }
+
+            return map_reduce(_children, [] (auto& child) {
+                return smp::submit_to(child.get_owner_shard(), [&child] {
+                    return child->get_progress();
+                });
+            }, progress{}, [] (progress l, progress r) {
+                progress res{};
+                res.total = l.total + r.total;
+                res.completed = l.completed + r.completed;
+                return res;
+            });
+
+        }
+
+        virtual std::string get_module() const = 0;
+
+        virtual is_abortable is_abortable() const {
+            return is_abortable::no;
+        }
+
+        void set_number(int number) {
+            _status.number = number;
+        }
+
+        void set_type(std::string type) {
+            _status.type = type;
+        }
+
+        void change_state(task_state state) {
+            _status.state = state;
+        }
+
+        void set_keyspace(std::string keyspace) {
+            _status.keyspace = keyspace;
+        }
+
+        void set_table(std::string table) {
+            _status.table = table;
+        }
+
+        void set_entity(std::string entity) {
+            _status.entity = entity;
+        }
+
+        void set_total_progress(double progress) {
+            _status.progress.total = progress;
+        }
+
+        void report_completion_progress(double progress) {
+            _status.progress.completed += progress;
+        }
+
+        void start(task_manager& tm) {
+            _status.start_time = db_clock::now();
+            _status.state = task_manager::task_state::running;
+
+            (void)with_gate(tm.async_gate(), [f = done(), &tm, id = _status.id] () mutable {
+                return std::move(f).finally([&tm, id] {
+                    return sleep_abortable(tm.task_ttl(), tm.abort_source());
+                }).finally([&tm, id] {
```

sleep_abortable throws `sleep_aborted` exception when aborted.
To ignore it we better use `.then_wrapped` here and call `f.ignore_ready_future()` to ignore the exception.
Otherwise, seastar will complain about ignored exception when this chain will be destroyed.